build/Targets/Imports.targets

```diff
@@ -2,7 +2,10 @@
 <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
-  
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
```

Comment:
Why is this needed?

Reply:
IIRC, changes to this file were not triggering an update to the build outputs. After a change was made to this file, its timestamp would end up newer than the output but CoreCompile wouldn't run so it would always be newer than the output. @rainersigwald should be able to provide more details about why this occurs.

Reply:
I'm still not sure how it got into the FUTD inputs (@panopticoncentral?), but that's the reason things never got up to date--FUTD was considering it out of date because `Imports.targets` had been modified since the last time the compiler ran, so a build was run, but then CoreCompile skipped because it wasn't listed in its "try to account for property changes by rebuilding when build logic changes" input:

https://github.com/dotnet/roslyn/blob/469fc641a6bafe92a81e1efae05ff8153906ee99/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L16-L17

Reply:
The current design of the up to date check is that we check to see if any of the imported target files have changed, assuming that if an imported target file has changed, the output might not be valid. This assumes that when we say its not up to date that MSBuild will agree and rebuild the project, so the outputs will be updated. If MSBuild isn't checking imported target files for up to date (and now that I'm writing this I'm kind-of remember having this discussion somewhere before), then I guess that logic is invalid and we should remove it since we'll get situations like this?

Reply:
What does this line do though?

Reply:
@jaredpar it marks the file as an input for CoreCompile, forcing a compilation to complete if it changes.

Reply:
Is there any file for which this shouldn't be added? Trying to think of an example. Possible if a file was VS design time only I guess?

Reply:
It's automatically added for **Directory.Build.props** and **Directory.Build.targets** for cases where the file is imported automatically by the project system. I'll add it for the others.

Reply:
Not quite what I'm asking. Still don't quite understand what this does. Trying to think of an example of a file where this wouldn't be needed. Will help me put this into context.

Reply:
@jaredpar It's not needed for most files inside a NuGet package, because packages of a fixed version are immutable. If the contents of the package changed, it means the version of the package changed which results in a change to the generated NuGet .props/.targets in the Obj directory, which *do* have these lines.

Reply:
@jaredpar It's needed in every imported MSBuild file and it should be done automatically and not manually. That is tracked as https://github.com/Microsoft/msbuild/issues/1299. For now, it should be done manually.

@panopticoncentral Yeah, I think we should consider backing out the import-tracking, or modifying it to be `$(MSBuildAllProjects)` based instead, to match the (bad) behavior of the build.